libexec/python/docker/api.py

```diff
@@ -51,13 +49,34 @@ def create_runscript(cmd,base_dir):
     output_file = write_file(runscript,content)
     return output_file
 
-def create_envfile(env, base_dir):
-    envfile = "%s/environment" % base_dir
+
+def create_envfile(base_dir,env=None,manifest=None):
+    '''create_envfile will generate a file called environment
+    in the base_dir of an image
+    :param base_dir: the base_dir to write the file to
+    :param manifest: the manifest to generate the Env for.
+    :param env: a list of environment variables to export, if manifest
+    is not provided
+    '''
+    if env == None:
+        if manifest == None:
+            logger.error("env or manifest must be provided, exiting")
+            sys.exit(1)    
+        env = get_config(manifest, 'Env').split("\n")
+
+    # A user providing an environment not split by '/n' --> error
+    if not isinstance(env,list):
+        env = [env]
```

Comment:
If the user (developer) gives a non-list (as the default field is returned as a string from the get_config function) then this will fail. This isn't a pretty line, but it makes sure that the function is robust.